projects/core/src/occ/utils/occ-asm-token.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpContextToken } from '@angular/common/http';
+
+export interface OccAsmContext {
```

Comment:
Perhaps the core side of the logic could be called something else than "Asm" specific names and make it more generic.  It's really just about sending a flag to ask for the user id to be added as a header.

I don't have specific names in mind, but it could be:
`UserIdHttpContext` or `CustomerIdHttpContext`

Reply:
The original idea was that this object could be extended for other flags needed for customer emulation. However, there is currently nothing else to send.
I am inclining toward your suggestion. Any thoughts @armanozak ?

Reply:
How about calling the interface just `OccHttpContext` and the token `OCC_HTTP_CONTEXT`? Maybe later we can even consider making the context interface augmentable.